R/module_teal_with_splash.R

```diff
@@ -56,17 +59,27 @@ ui_teal_with_splash <- function(id,
 #' If data is not loaded yet, `reactive` returns `NULL`.
 #' @export
 srv_teal_with_splash <- function(id, data, modules, filter = teal_slices()) {
-  checkmate::assert_multi_class(data, c("TealDataAbstract", "teal_data"))
+  checkmate::check_multi_class(data, c("TealData", "teal_data", "teal_data_module"))
+
   moduleServer(id, function(input, output, session) {
-    logger::log_trace("srv_teal_with_splash initializing module with data { toString(get_dataname(data))}.")
+    logger::log_trace("srv_teal_with_splash initializing module with data.")
 
     if (getOption("teal.show_js_log", default = FALSE)) {
       shinyjs::showLog()
     }
 
     # raw_data contains teal_data object
     # either passed to teal::init or returned from ddl
-    raw_data <- if (inherits(data, "teal_data")) {
+    raw_data <- if (inherits(data, "teal_data_module")) {
+      do.call(
+        data$server,
+        append(
+          list(id = "teal_data_module"),
+          attr(data, "server_args") # might be NULL or list() - both are fine
+        ),
+        quote = TRUE
+      )
```

Comment:
```suggestion
      data$server(id = "teal_data_module")
```

Server function must only take `id`, right?

Reply:
true, this is a remaining from the previous implementation. I'll fix this